pkg/minikube/cluster/cluster.go

```diff
@@ -140,11 +144,23 @@ func DeleteHost(api libmachine.API) error {
 
 // Un-register host VM
 func unregister(host *host.Host) error {
+	if !minishiftConfig.InstanceConfig.Register {
+		return nil
```

Comment:
Same as above, it is unfortunate to create a dependency to minishiftConfig.InstanceConfig.Register.